### DISCUSSION TITLE
G7 "Manual Slot Number"

### DISCUSSION THREAD

**Discussion by jbowler** [NightscoutFoundation/xDrip#3524] [Q&A]
I've used the G7 manual slot number (the engineering mode option) pretty much from the start with the G7.  I set the manual slot number on my primary 'phone to "1" (i.e. the receiver slot) and use slot 2 if I need to connect from another 'phone, in particular if I need to use the Dexcom app to upload information.

This has worked fine until, on May 27 for the first time in a long time (months) I had to start a new sensor with the Dexcom app (so that got slot 2).  When I tried to connect my 'phone using slot 1 I got repeated "mismatch" errors; I got authentication reports on the status screen but no "pair" pop-up and then the "mismatch" error.

Eventually I shut down the other 'phone, waited, and swapped to slot 2; xDrip+ connected just fine.

Today I started the next sensor from my burner 'phone with the Dexcom app (I'm adopting this as the new default), waited for the old sensor to expire then connected from my new 'phone with slot 1.  Same problems again.  I shut down the burner, waited a little, and tried again (still on slot 1 this time) and it all worked just fine!

I looks like "manual slot number" is broken in xDrip+; it looks like it is always trying "slot 2" even though I set it to "slot 1".  I can't see a "discussion" for this; any ideas?

I sent the log which should include the relevant events.

**Comment by Navid200** [comment]
The manual slot number change feature was never approved.  That's why it is still behind engineering mode:
https://navid200.github.io/xDrip/docs/G6_slot.html

There is no need for it.  I have been using G7 for 8 months with xDrip.  I start a next sensor as I have explained here:
https://navid200.github.io/xDrip/docs/Dexcom/SubsequentG7.html
I never need to change the slot number.

Please tell me why you need it.  Or tell me what you need to do that you think requires it.

**Reply by jbowler** [reply]
> The manual slot number change feature was never approved. That's why it is still behind engineering mode: https://navid200.github.io/xDrip/docs/G6_slot.html

Ok, so you are confirming that even though I have engineering mode enabled and I have the xDrip+ setting set to "1" it is being ignored and slot number 2 is being used.

Have I got that right?

**Reply by Navid200** [reply]
I am telling everyone that they should not be using engineering mode except for performing tests if they are xDrip developers.
The engineering mode is not just a compartment for additional settings for the users to use when they wish.

A company (CamAPS) came up with a pump that used the mobile app slot instead of using the medical slot.  Imagine if a car manufacturer comes up with a car with the gas pedal on the left and the brake pedal on the right!  
We decided to offer a solution for the mess created by this company (Bluetooth slot change).  The attempt failed.  Hence, the setting was never removed from the engineering mode and very likely will be completely removed soon.  
We then came up with the companion app mode approach, which has been a success.

There is no reason to change the Bluetooth slot used by xDrip.  If you do, anything could happen as the outcome is not predictable with all handsets.  So, yes, it is possible that the setting may be ignored.  But, it is really not a safe setting to use.

**Reply by jbowler** [reply]
> So, yes, it is possible that the setting may be ignored.

Well, if it *isn't* "being ignored" then there is a bug which should be fixed because it is sitting there in the (engineering mode) UI and if it *is* "being ignored" then the ability to set it (and have it ignored) should not be clearly marked *IGNORED*.

But everything else you have said is completely off topic.  Perhaps you could check if it is being ignored before launching into all this off topic stuff?  So far as I can tell it is.

**Reply by Navid200** [reply]
Tell me why you need to change the Bluetooth slot.  There may be a workaround.
I want to help you use G7.  

I also like to know if there is any problem with xDrip using G7 outside engineering mode.  
You will be helping us, volunteers, improve xDrip.

**Comment by jbowler** [comment]
Using two 'phones one with the Dexcom G7 app installed allows me to get information to Dexcom for customer support without disrupting my normal non-Dexcom use of the G7.  If they both use sensor slot 2 I have to disconnect a problem sensor from xDrip+ by shutting down the 'phone (so stopping AndroidAPS until the Dexcom app has picked up the sensor.)

Activating with the Dexcom app means I have a record of the date and ID of the sensor application (so I can use the on-line failed sensor reporting) and I have a separate monitor of start-up.  In event of a complete failure I have the information for Dexcom; in my May 27 case customer service only have a record of one sensor (the one I activated with their app.)

Two 'phones with two sensors allows me to see a failed sensor during the grace period.  If both sensors use slot 2 (and this seems to be the case) I have to remember to shut down the Dexcom 'phone half an hour (or so) before the grace period ends.  This is possible and this is what I will do if the "manual" slot number feature has been removed.

Note that I don't change the manual slot number; I always have it set on "1" and I have since I started using the G7 before AndroidAPS 3.2 was released.  Since I don't have a receiver this is just the simplest most flexible approach.

**Reply by Navid200** [reply]
Thanks

You bring up a very good point about reminders.  We currently have a sensor expiry notification option:
https://navid200.github.io/xDrip/docs/Sensor-Expiry.html
But, it gives notifications at 24, 9 and 2 hours before expiry.  So, for a G7, the last notification will show 2 hours before the end of 10.5 days.
I have experienced situations where the sensor has stopped working and I did not switch right away.  
I will open a submission to add an additional notification exactly when the sensor expires.
Thanks

I now understand why you like to have another phone that can connect to your G7 simultaneously.
I can think of a workaround.  You don't have to use it.  But, please hear me out.
You can do everything as you are doing now except using your xDrip out of engineering mode.  The result is that both xDrip and Dexcom ap will attempt to connect to the same slot and one will fail if you do this at the same time.
Please hear me out:  
Once a day, put your xDrip phone away or in the microwave or turn it off for only 15 minutes.  At the same time, turn on your Dexcom app phone.  G7 backfills 24 hours.  So, if you do this everyday at the same time, Dexcom app will have everything in it as it would if you kept it on all the time.

So, you never use both Dexcom app and xDrip at the same time.  But, since G7 backfills 24 hours, it is enough to enable Dexcom app only once a day when you temporarily disable xDrip.

I know this is not as convenient as what you wish you could do.  But, the option to change the Bluetooth slot did not provide predictable results with the few phones brands and models we tested.  
If someone can figure out a way to fix it, I will support it and I will give them a hand if they need it.
But, it is possible this cannot be fixed.  it is not because any of us has decided not to fix it.  
Some things work reasonably well on most phones.  Some don't.  

This is all I can suggest.
And everyone else is welcome to provide suggestions. 

Thanks

**Reply by jbowler** [reply]
> I now understand why you like to have another phone that can connect to your G7 simultaneously.
> I can think of a workaround.

Your work-round doesn't permit simultaneous connection any more than mine does and, indeed, while some people may need to have all their data stored on Dexcom servers I don't.  I only need sufficient data for Dexcom to do a replacement at least without using my three/year "no questions asked" slots, maybe not even then; I've found xDrip+ with the G7 reliable.  The two sensors that triggered this were physical failures caused by problems with the inserter.

We-all have no problem getting two simultaneous connections on slot 2 so long as we use separate 'phones and separate **transmitters**.  I can arrange to do that by using my burner to just start the sensor.  At this point Dexcom knows I did it and that I'm not trying to rip them off with a sensor serial number I got from someone else (so I'm not going to have to return the sensor if it goes bad.)  I suspect this is sufficient for them and I admit it's reasonable.

The only time I may need a simultaneous connection is if I get a wonky sensor that **hasn't** failed; if it has failed I replace on my main 'phone so can reconnect the burner (simultaneous, two different sensors/transmitters).   All the same that is not a big deal and it hasn't happened to me so far.

In other words I have a work-round that works and doesn't require me to disable AndroidAPS for half an hour.

Yet it is still a work-round; I can use it _if the manual slot facility has been removed_ but I'd rather just have the old system where I could make xDrip+ use the receiver slot.  This is what this discussion is about; I can (and will) live with it if you have removed "manual slot number" but I'd like to know.

BTW if you want to prevent people like me from using "engineering mode" then you should stop documenting it, change the access mechanism and remove all vestiges from the UI.  I'm not an engineer but I am an extremely experienced software developer as you know because you can read my profile and, for that matter, you can see my comments and see that I have even less sympathy for time wasters than you do.

**Reply by Navid200** [reply]
There is nothing wrong with using engineering mode when used for the right reason.
The documentation is there to explain what it is for and how to use it correctly.
